### DISCUSSION TITLE
Pyswmm controls does not change SWMM control rules

### DISCUSSION THREAD

**Discussion by lukc19** [pyswmm/pyswmm#421] [Q&A] [answered]
The following code does not change SWMM control rules; the output file that I am able to access after simulation shows no changes There aren't any rules in the [CONTROLS] section of the model.

    from pyswmm import Simulation, Links, Nodes
    from datetime import datetime,timedelta
    control_time_step = =900
        with Simulation(tmp_file) as sim:
            sim.step_advance(control_time_step)
            node_object = Nodes(sim) 
            l1 = node_object["SJ1"]
            l2 = node_object["SJ2"]
            l3 = node_object["SJ3"]
            l4 = node_object["SJ4"]
            l5 = node_object["SJ5"]

            link_object = Links(sim) 
            v1 = link_object["OR1"]
            v2 = link_object["OR2"]
            v3 = link_object["OR3"]
            v4 = link_object["OR4"]

            v1.target_setting = 1 
            v2.target_setting = 1
            v3.target_setting = 1
            v4.target_setting = 1

     
            for step in sim:
                if l5.depth > 0.2:
                    v2.target_setting = 0
                    v3.target_setting = 0
            
          sim.close()
          sim.execute()
[swmm_run_rbc.zip](https://github.com/OpenWaterAnalytics/pyswmm/files/11171160/swmm_run_rbc.zip)

**Comment by bemcdonnell** [comment, marked as answer]
@lukc19 for starters, inside the `with` context manager, delete the `sim.close()` and `sim.execute()`.  Give that a quick try

**Reply by lukc19** [reply]
It works! thank u so much!!!

**Reply by bemcdonnell** [reply]
@lukc19 you're welcome!  For reference here is the documentation showing the different methods you can run a simulation http://wateranalytics.org/pyswmm/quickstart.html :)
